tests/unit/test-util/test.expand-prefs.js

```diff
@@ -0,0 +1,68 @@
+import { assert, expect } from 'chai';
+import { describe, it } from 'mocha';
+
+import expandPrefs from '../../../src/util/expand-prefs.js';
+
+describe('utils/expand-prefs', () => {
+  it('expands dot-deliminated preferences into a deep object', () => {
+    const input = {
+      a: 'a',
+      'b.c': 'c',
+      'd.e.f': 'f',
+    };
+    const expected = {
+      a: 'a',
+      b: {
+        c: 'c',
+      },
+      d: {
+        e: {
+          f: 'f',
+        },
+      },
+    };
+    const actual = expandPrefs(input);
+
+    assert.deepEqual(actual, expected);
+  });
+
+  it("doesn't pollute the object prototype", () => {
+    const call = 'overriden';
+    const input = {
+      'hasOwnProperty.call': call,
+    };
+    const expected = {
+      hasOwnProperty: {
+        call,
+      },
+    };
+    const actual = expandPrefs(input);
+
+    assert.notEqual(Object.prototype.hasOwnProperty.call, call);
```

Comment:
Instead of notEqual, save the original value before modifying the test, then do a strict comparison. A unit test that tests for equality is stronger than one that tests for inequality.